filament/backend/src/opengl/OpenGLProgram.cpp

```diff
@@ -203,6 +210,29 @@ void OpenGLProgram::initializeProgramState(OpenGLContext& context, GLuint progra
         }
     }
     mUsedBindingsCount = usedBindingCount;
+
+    // Push constant initialization
+    auto& constants = lazyInitializationData.pushConstants;
+    if (!constants.empty()) {
+        mVertexPushConstants.reserve(constants.size());
+        mVertexPushConstants.resize(constants.size());
+        // The constants are defined in a struct of name PUSH_CONSTANT_STRUCT_VAR_NAME. We prepend
+        // the prefix here to avoid string manipulation in a the draw loop.
+        std::stringbuf constantPrefix;
+        constantPrefix.sputn(Program::PUSH_CONSTANT_STRUCT_VAR_NAME,
+                strlen(Program::PUSH_CONSTANT_STRUCT_VAR_NAME));
+        constantPrefix.sputn(".", 1);
+
+        std::transform(constants.begin(), constants.end(), mVertexPushConstants.begin(),
+                [&constantPrefix, program](Program::PushConstant& constant) -> PushConstantBundle {
+                    constant.name.insert(0, utils::CString(constantPrefix.str().c_str()));
+                    auto const loc = glGetUniformLocation(program, constant.name.c_str());
+                    return {
```

Comment:
you need to check if loc is < 0, that would happen if somehow the uniform wasn't found. I think that can happen if the uniform is not used by the shader.  So this is not an error. It just means that setting the push constant will be a no-op.

Reply:
done in OpenGLDriver